client/src/client.rs

```diff
@@ -14,7 +14,7 @@ use config::{CheckpointFallback, Config};
 use consensus::{types::Header, ConsensusClient};
 use execution::types::{CallOpts, ExecutionBlock};
 use log::{error, info, warn};
-use tokio::sync::RwLock;
+use tokio::sync::{Mutex, RwLock};
```

Comment:
As there is no need to persist the mutex across awaits I think we can use lighter `std::sync::Mutex`